### DISCUSSION TITLE
18 gennaio 2022 A - esercizio 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#2] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/bab36557-4d63-4158-9238-434a41238ed6)

**Comment by IonutCicio** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/262e3e9b-6f41-4651-b926-21583c64910a)

**Reply by Frisbon** [reply]
Hola, una curiositá.
Ti sei subito ricavato gli aerei **non revisionati** e ti sei estratto solamente l'ID.
Poi ti sei ricavato tutti i voli di **tutti gli **aerei**** che so volati a NY nel 2021.
Infine hai fatto la join con questi voli e... la tabella di tutti gli aerei? 
Magari un typo, suppongo volessi usare Aerei_Mai_Revisionati invece che Aereo nell'ultima join.
Se fosse cosí, non mi convince perché ti sei preso solo l'ID nella prima riga, come fai a proiettare sul Modello nella query finale?

**Reply by IonutCicio** [reply]
Hai ragione 👍
Avrei dovuto scrivere $\text{VIAGGI-2021-NY} \bowtie \text{Aereo} \bowtie \text{AEREI-MAI-REVISIONATI}$, oppure **non fare la proiezione** su $\text{ID}$ e fare direttamente $\text{VIAGGI-2021-NY} \bowtie \text{AEREI-MAI-REVISIONATI}$

**Comment by niccolo-fato** [comment]
![2024-01-13 12 05 25](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/10088024-947a-4347-b21d-e78a17511d69)

**Comment by Frisbon** [comment]
la 1b) mi viene cosí
perfavore correggetemi, non le so fare
![image_2024-01-14_10-42-19](https://github.com/sapienzastudentsnetwork/bdi2324/assets/84731341/58ccf731-0ce2-4934-ad11-18b381e0a939)

**Reply by IonutCicio** [reply]
Hai scritto $\underset{\text{Città } = \text{ Roma}}{\sigma}$ 2 volte. Attenzione a **non** partizionare su $\text{Nome}$ e $\text{Cognome}$! Partiziona **sempre sull'ID**, e poi fai una join alla fine per ottenere i dati che ti servono, altrimenti potresti avere problemi di omonimia. Per il resto pare andare bene.

**Reply by Elia-Belli** [reply]
Per il **Personale Totale** non basterebbe:  $\pi_{(ID)}(Personale)$
